signer/migrations/0003__create_tables.sql

```diff
@@ -184,3 +191,111 @@ CREATE TABLE sbtc_signer.withdrawal_reject_events (
     created_at    TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP NOT NULL
 );
 
+-- Represents a combined transaction package which is broadcasted to the Bitcoin
+-- network. The transaction package is built up of multiple transactions which
+-- are tracked separately in the `sweep_transactions` table. A transaction
+-- package may contain the sweeping transactions for both deposit and
+-- withdrawal requests.
+CREATE TABLE sbtc_signer.sweep_packages (
+    -- Internal ID of the package
+    id SERIAL PRIMARY KEY,
+    -- The Bitcoin block hash at which this package was created.
+    created_at_block_hash BYTEA NOT NULL,
+    -- The Bitcoin market fee rate at the time this package was created.
+    market_fee_rate DOUBLE PRECISION NOT NULL,
+    -- Timestamp of when this package was created.
+    created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+
+-- Represents an individual transaction within a broadcasted sweep transaction
+-- package. Individual deposit and withdrawal requests reference back to these
+-- transactions to keep track of both the overall transaction package as well as
+-- the individual Bitcoin transactions they are related to.
+CREATE TABLE sbtc_signer.sweep_transactions (
+    -- Internal ID of the transaction.
+    id BIGSERIAL PRIMARY KEY,
+    -- The ID of the grouping transaction package.
+    sweep_package_id INTEGER NOT NULL,
+    -- The Bitcoin transaction ID of the transaction.
+    txid BYTEA NOT NULL,
+    -- The signer UTXO being spent in this transaction.
+    utxo_txid BYTEA NOT NULL,
+    utxo_output_index INTEGER NOT NULL,
+    -- The total amount of the transaction.
+    amount BIGINT NOT NULL,
+    -- The fee paid for the transaction.
+    fee BIGINT NOT NULL,
+    -- The fee rate in satoshis per vByte used for this transaction.
+    fee_rate DOUBLE PRECISION NOT NULL,
```

Comment:
But we could note the market fee rate. That information is good for diagnostic purposes.

Reply:
I was originally down a similar path but felt it cost more in code than it was worth. This is provided as an `f64` from `estimate_fees()` (or similar) so I think there's greater risk in losing "data" in the otherwise conversion.

The market fee rate is on the package-level (covering all transactions).

Reply:
I'm not suggesting that we use the fee rate to compute the transaction's size, but get and store the size from the actual transaction.

Reply:
Yeah we could do that, it's not a problem, but what I meant is that this is the literal value that comes from the `SignerBtcState` associated with the transaction itself.

Reply:
Oh I see. Let me start by acknowledging that my original reason for why I thought that we shouldn't store the fee rate was not compelling. I have a different suggestion, and it allows us to leave this column out if we want:

* we should store the `vsize` for each transaction
* we should store the `vsize` for each input and output linked to a request in the transaction.

This allows us to compute the assessed fee for all inputs and outputs. It also allows us to compute the fee rate for the entire transaction if we want to.

---

About using the `SignerBtcState` struct, that only has the market fee rate. I don't think any of our structs contain the fee rate for the bitcoin transaction.

Unrelated, just wanted to call out that while each transaction targets the market rate, the actual fee rate for any transaction can be quite a bit more than the market rate if we are RBFing a transaction.

Reply:
Do we really want to store the `vsize` of every input & output? Isn't it enough to have it on the transaction? I'm thinking that if we know our sats/vbyte and the previn fee then we should have everything we need to ensure that both the fee and rate are high enough in an eventual RBF.. but maybe I'm thinking wrong?

Reply:
> Do we really want to store the `vsize` of every input & output? Isn't it enough to have it on the transaction?

Oh, you're right, we can compute what is necessary when we grab the raw transaction.


> I'm thinking that if we calculate our sats/vbyte on the tx-level from `vsize` and `fee` then we should have everything we need to ensure that both the fee and rate are high enough in an eventual RBF.. but maybe I'm thinking wrong?

I was thinking about using this information for validating the stacks `complete-deposit` contract call. There we ensure that the assessed fee is under the request's max-fee, and for that we use the `vsize` of all request inputs and outputs.